step-ca/certificate-authority-server-production.mdx

```diff
@@ -175,112 +174,173 @@ So do some certificates.
-
-While `step-ca` doesn't directly support active revocation mechanisms like Certificate Revocation Lists (CRLs) or the Online Certificate Status Protocol (OCSP),
-you can independently manage your own CRL if you like.
-
-#### Create an intermediate CA with a CRL distribution endpoint
-
-Let's make it possible to revoke your intermediate CA down the road if necessary.
-This setup is more complex than the default `step-ca` PKI, but it offers an insurance policy for a compromised intermediate CA.
-
-1. Create an intermediate CA that includes a CRL endpoint. Save the following template to `intermediate.tpl`:
-
-  ```json
-  {
-          "subject": {{ toJson .Subject }},
-          "keyUsage": ["certSign", "crlSign"],
-          "basicConstraints": {
-                  "isCA": true,
-                  "maxPathLen": 0
-          },
-          "crlDistributionPoints":
-                  ["http://crl.example.com/crl/ca.crl"]
-          }
-  }
-  ```
+You could issue a new intermediate using your root CA key,
+but your old intermediate had a 10 year validity period!
+So, you're stuck having to rotate your root CA too,
+and that may be a big project:
+you have to distribute the new root certificate to clients,
+and ensure the old one is no longer trusted.
 
-  You'll need this template to manually create your intermediate CA.
-  The CRL endpoint here should be an HTTP URL; the CRL file itself is signed.
-  The CRL will be a static file, so you you might choose an object storage or CDN endpoint here.
-  
-  Use the template to create your intermediate CA. You will need your root CA certificate and key:
-  
-  ```bash
-  $ step certificate create \
-       --template intermediate.tpl \
-       --ca $(step path)/certs/root_ca.crt \
-       --ca-key $(step path)/secrets/root_ca_key \
-       --not-after 87660h \
-       "Example Intermediate CA" \
-       $(step path)/certs/intermediate_ca.crt \
-       $(step path)/secrets/intermediate_ca_key
-  ```
+To avoid this scenario, you can use _active revocation_,
+making it possible to immediately revoke a compromised certificate.
 
-2. Create an empty CRL file and sign it with your root CA key:
+Active revocation can also be used on leaf certificates.
+If a long-lived leaf certificate is compromised,
+it can be rendered unusable by an attacker through revocation.
 
-  ```bash
-  cat <<EOF > openssl.conf
-  [ ca ]
-  default_ca = CA_default
-  
-  [ CA_default ]
-  default_crl_days = 30
-  database = index.txt
-  default_md = sha256
-  EOF
-  touch index.txt
-  openssl ca \
-      -config openssl.conf \
-      -gencrl \
-      -keyfile $(step path)/secrets/root_ca_key \
-      -cert $(step path)/certs/root_ca.crt \
-      -out ca.crl.pem
-  openssl crl \
-      -inform PEM \
-      -in ca.crl.pem \
-      -outform DER \
-      -out ca.crl
-  ```
+But there are downsides:
+CRL adds a service dependency to your PKI.
+Clients check the CRL endpoint on every new connections,
```

Comment:
```suggestion
Clients check the CRL endpoint on every new connection,
```

Reply:
Technically clients could be (and probably are) caching the CRL for a specific duration. The CRL also contains an indication as to when the new CRL can be expected. It's possible it happens earlier, but it does put a limit on how the period that an outdated CRL could be in use. There are also delta CRLs, but I don't think we support that.

OCSP happens on every request, or it gets stapled in the TLS response from a server.